File: python-automation/src/automation/pages/checkout_page.py
from playwright.sync_api import Page
from src.automation.data.test_data import Urls


class CheckoutPage:
    def __init__(self, page: Page):
        self.page = page

    @property
    def first_name_input(self):
        return self.page.locator("[data-test='firstName']")

    @property
    def last_name_input(self):
        return self.page.locator("[data-test='lastName']")

    @property
    def postal_code_input(self):
        return self.page.locator("[data-test='postalCode']")

    @property
    def continue_button(self):
        return self.page.locator("[data-test='continue']")

    @property
    def finish_button(self):
        return self.page.locator("[data-test='finish']")

    @property
    def error_message(self):
        return self.page.locator("[data-test='error']")

    @property
    def complete_header(self):
        return self.page.locator("[data-test='complete-header']")

    def fill_customer_info(self, first_name: str, last_name: str, postal_code: str):
        self.first_name_input.fill(first_name)
        self.last_name_input.fill(last_name)
        self.postal_code_input.fill(postal_code)

    def continue_checkout(self):
        self.continue_button.click()

    def finish_order(self):
        self.finish_button.click()
        assert self.page.url == Urls.CHECKOUT_COMPLETE, \
            f"User should be redirected to {Urls.CHECKOUT_COMPLETE}, but got {self.page.url}"

    def assert_step1_error(self, expected_text: str):
        error_text = self.error_message.text_content()
        assert expected_text in error_text, \
            f"Error message should contain: {expected_text}, but got {error_text}"

File: java-automation/src/main/java/com/automation/pages/CheckoutPage.java
package com.automation.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.Locator;
import com.automation.data.TestData;

import static org.junit.jupiter.api.Assertions.*;

public class CheckoutPage {
    private final Page page;

    public CheckoutPage(Page page) {
        this.page = page;
    }

    private Locator getFirstNameInput() {
        return page.locator("[data-test='firstName']");
    }

    private Locator getLastNameInput() {
        return page.locator("[data-test='lastName']");
    }

    private Locator getPostalCodeInput() {
        return page.locator("[data-test='postalCode']");
    }

    private Locator getContinueButton() {
        return page.locator("[data-test='continue']");
    }

    private Locator getFinishButton() {
        return page.locator("[data-test='finish']");
    }

    private Locator getErrorMessage() {
        return page.locator("[data-test='error']");
    }

    public Locator getCompleteHeader() {
        return page.locator("[data-test='complete-header']");
    }

    public void fillCustomerInfo(String firstName, String lastName, String postalCode) {
        getFirstNameInput().fill(firstName);
        getLastNameInput().fill(lastName);
        getPostalCodeInput().fill(postalCode);
    }

    public void continueCheckout() {
        getContinueButton().click();
    }

    public void finishOrder() {
        getFinishButton().click();
        assertEquals(TestData.Urls.CHECKOUT_COMPLETE, page.url(), 
            "User should be redirected to checkout complete page");
    }

    public void assertStep1Error(String expectedText) {
        assertTrue(getErrorMessage().textContent().contains(expectedText), 
            "Error message should contain: " + expectedText);
    }
}
